### PR TITLE
✨ Profile settings

### LINKED ISSUE #64: Profile settings
We need to add the same Manage profile settings that we have in the mobile app.

### PR DESCRIPTION
**Pull request by jozsefsallai**
closes #64 

ZZ5ULPAcXd

todo list

- [x] toggle followers count
- [x] toggle community posts
- [x] change visibility
- [x] change basic details
- [x] change profile picture and cover photo
- [x] i18n
- [x] proper error handling

### CODE REVIEW

**Comment by lifenautjoe** [comment]
Thanks Joe. Could you add some screens for historical and review purposes too? 🙏🏼

**Comment by jozsefsallai** [comment]
gif is too large so I had to upload it to imgur :) here's how avatar/cover changing works: https://i.imgur.com/xs1D2ET.mp4

**Comment by jozsefsallai** [comment]
Server-side errors are handled now using toasts:

Or8dbmXn5E

**Comment by lifenautjoe** [comment]
Some comments on the design

* Please make inputs 100% of width of the line they're currently in and make them circular like in the login page
* Make the edit and delete icon buttons circular and add more padding, check the mobile app for reference. Show only one at a time. So if theres a pic, show the trash, after the pic was "deleted" on the UI, show the edit button. 
* Move the save buttons to the bottom right and the cancel to the bottom left
* The profile and cover photo shouldnt actually be changed until the save button is pressed

**Comment by jozsefsallai** [comment]
There's client-side validation now as well, for the name and the username:

jrAQasNQS6

> Please make inputs 100% of width of the line they're currently in and make them circular like in the login page

The reason I did not do that is that the mobile app doesn't have it like that:

image

I moved the icon to the top because I thought it looked pretty bad with our current setup when centered:

image

...but I can make the requested changes :)

> So if theres a pic, show the trash, after the pic was "deleted" on the UI, show the edit button.

I agree that there is no need for a trash button when there is no image - I will remove it. But I feel like it would be more intuitive if both buttons were present when there is an image - one to change it and one to remove it. Instead of having to remove it first and only then being able to change it. What is your opinion on that?

> The profile and cover photo shouldnt actually be changed until the save button is pressed

I think I didn't do that because of the limitations of our current modal service system. But I'll see if I can implement something like that. Might be possible by passing props back and forth.

**Comment by jozsefsallai** [comment]
New UI:

image

**Comment by jozsefsallai** [comment]
I have implemented the last change regarding avatars/cover photos. Now instead of updating the avatar/cover when the save button is pressed in the cropping modal, it will send the data URL as well as the blob object back to the user profile settings modal and updates the new values in the form's state, so the images will only be updated when the "Save" button is pressed on the user profile settings modal. Here's a demo on how it works (on imgur again, the file size is literally 0.7 MB larger than what github allows): https://i.imgur.com/0UfTO4a.mp4 

Like I said in my previous comment, the current way of managing modals doesn't allow easy state transfer between multiple modals, so I had to pass props to the two modals in a bidirectional fashion. I think there is nothing wrong with this approach, but we might want to improve this when/if we improve modals with pagination support.

**Comment by lifenautjoe** [comment]
Looks good! Let's also only show one button at a time on the cover and avatar. I understand it is useful to have both options immediately but design wise, looks a bit funky and also out of sync with the mobile app version. 

Finally, add a bit more padding to those icon buttons like the mobile app, I think like 10 - 15px and we're good to go!

File: components/settings/user-settings/OkUserVisibilitySettings.vue [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        mounted() {
+            if (this.userService.loggedInUser) {
+                const { visibility } = this.userService.loggedInUser.value;
+                this.visibility = visibility;
+            }
+        }
+
+        async setUserVisibility(visibility: UserVisibility) {
+            this.visibility = visibility;
+            this.saving = true;
```
Comment (lifenautjoe) [inline components/settings/user-settings/OkUserVisibilitySettings.vue:117]: Its fine for now but for future reference, please use. the `is` prefix ( `isSaving` ) for boolean variables .

**Review by lifenautjoe — APPROVED** [review]

**Pull request merged**